### ISSUE TITLE
AScan API - Allow to scan as an user

### ISSUE THREAD

**Issue by zapbot** [zaproxy/zaproxy#1621]
```
From Issue 1614.

Add an action to the AScan API that allows to scan as an user.
The action will be named "scanAsUser" (the same as the spider) and will allow to specify
the user with the following mandatory parameters:
contextId
userId

It will also contain the following parameters (the same as the "scan" action):
url (mandatory)
recurse
inScopeOnly
scanPolicyName
method
postData
```

Original issue reported on code.google.com by `THC202` on 2015-05-02 15:32:22

**Comment by thc202** [comment]
Fixed in fad38cc6a5a931ebd33bb1e265793f929c2ed79d

**Comment by thc202** [comment]
Released in version 2.4.1.
